aspnetcore/tutorials/signalr-typescript-webpack.md

```diff
@@ -0,0 +1,245 @@
+
+## Prerequisites
+
+Install the following software:
+
+# [Visual Studio](#tab/visual-studio)
+
+* [.NET Core SDK 2.1 or later](https://www.microsoft.com/net/download/all)
+* [Node.js](https://nodejs.org/) with [npm](https://www.npmjs.com/)
+* [Visual Studio 2017](https://www.visualstudio.com/downloads/) version 15.7 or later with the **ASP.NET and web development** workload
+
+# [.NET Core CLI](#tab/netcore-cli)
+
+* [.NET Core SDK 2.1 or later](https://www.microsoft.com/net/download/all)
+* [Node.js](https://nodejs.org/) with [npm](https://www.npmjs.com/)
+
+---
+
+## Create the ASP.NET Core web app
+
+# [Visual Studio](#tab/visual-studio)
+
+Configure Visual Studio to look for npm in the *PATH* environment variable. By default, Visual Studio uses the version of npm found in its installation directory. Follow these instructions in Visual Studio:
+
+1. Navigate to **Tools** > **Options** > **Projects and solutions** > **Web Package Management** > **External Web Tools**.
+1. Select the *$(PATH)* entry from the list. Click the up arrow to move the entry to the second position in the list. As an aside, the first entry refers to the project's local packages.
+
+    ![Visual Studio Configuration](signalr-typescript-webpack/_static/signalr-configure-path-visual-studio.png)
+
+Visual Studio configuration is completed. It's time to create the project.
+
+1. Use the **File** > **New** > **Project** menu option and choose the **ASP.NET Core Web Application** template.
+1. Name the project *SignalRWebPack*, and click the **OK** button.
+1. Select *.NET Core* from the target framework drop-down, and select *ASP.NET Core 2.1* from the framework selector drop-down. Select the **Empty** template, and click the **OK** button.
+
+# [.NET Core CLI](#tab/netcore-cli)
+
+Run the following command in the **Integrated Terminal**:
+
+```console
+dotnet new web -o SignalRWebPack
+```
+
+An empty ASP.NET Core web app, targeting .NET Core, is created in a *SignalRWebPack* directory.
+
+---
+
+## Configure Webpack and TypeScript
+
+The following steps configure the conversion of TypeScript to JavaScript and the bundling of client-side resources.
+
+1. Execute the following command in the project root to create a *package.json* file:
+
+    ```console
+    npm init -y
+    ```
+
+1. Add the highlighted property to the *package.json* file. It prevents package installation warnings in the next step.
+
+    [!code-json[package.json](signalr-typescript-webpack/sample/snippets/package1.json?highlight=4)]
+
+1. Install the required npm packages. Execute the following command from the project root:
+
+    ```console
+    npm install -D -E clean-webpack-plugin@0.1.19 css-loader@0.28.11 html-webpack-plugin@3.2.0 mini-css-extract-plugin@0.4.0 ts-loader@4.4.1 typescript@2.9.2 webpack@4.12.0 webpack-cli@3.0.6
+    ```
+
+    Exact versions of npm packages, which are known to work, are installed.
```

Comment:
"Exact versions of npm packages, which are known to work, are installed."
I'm unsure what "exact versions" means? The latest stable version of the package? Just the latest? Something something?
Also, is it necessary to point out that they're known to work?

Reply:
@rachelappel Does it sound clearer now? The idea is to prevent surprises when the reader is walking through the tutorial. I want to ensure they're using the same package versions that were known to work for me.

Reply:
@scottaddie Yeah that sounds better